java/configuration/src/main/java/sleeper/configuration/properties/UserDefinedInstanceProperty.java

```diff
@@ -178,6 +178,7 @@ public enum UserDefinedInstanceProperty implements InstanceProperty {
     COMPACTION_JOB_CREATION_LAMBDA_TIMEOUT_IN_SECONDS("sleeper.compaction.job.creation.timeout.seconds", "900", Utils::isValidLambdaTimeout),
     MAXIMUM_CONCURRENT_COMPACTION_TASKS("sleeper.compaction.max.concurrent.tasks", "300"),
     COMPACTION_TASK_CREATION_PERIOD_IN_MINUTES("sleeper.compaction.task.creation.period.minutes", "1"), // >0
+    COMPACTION_TASK_CPU_ARCHITECTURE("sleeper.compaction.task.cpu.architecture", "ARM64"),
     COMPACTION_TASK_CPU("sleeper.compaction.task.cpu", "2048"),
     COMPACTION_TASK_MEMORY("sleeper.compaction.task.memory", "4096"),
```

Comment:
I think we should replace the properties `sleeper.compaction.task.cpu` and `sleeper.compaction.task.memory` with two specific ones, one for ARM, one for x86. That will let us set good defaults for the different architectures.

Reply:
I'll also check whether the current default CPU value makes any difference on x86, comparing 1024 to 2048 (1 vs 2 CPUs).

Reply:
I've run one system test for each of these configurations. Here are the results for compaction:

2 CPUs, ARM:
Ran 1 standard, 8 splitting
Average standard compaction rate: 237,823.72 records/s
Average splitting compaction rate: 330,255.25 records/s

1 CPU, ARM:
Ran 1 standard, 7 splitting
Average standard compaction rate: 232,176.25 records/s
Average splitting compaction rate: 292,021.27 records/s

2 CPUs, x86:
Ran 1 standard, 7 splitting
Average standard compaction rate: 336,100.05 records/s
Average splitting compaction rate: 455,389.67 records/s

1 CPU, x86:
Ran 2 standard, 8 splitting
Average standard compaction rate: 352,782.74 records/s
Average splitting compaction rate: 420,589.43 records/s

Reply:
Some of those numbers aren't as comparable because the compaction happened to trigger at a time that resulted in extra jobs being created, and jobs with fewer records have a higher compaction rate than jobs with more records.

That also results in more files for the splitting compaction to process which might throw the numbers off a bit.

I wish we could make this more deterministic. I don't think repeating these tests will be enough to give really comparable data.

Reply:
[report-arm-1cpu.txt](https://github.com/gchq/sleeper/files/9960033/report-arm-1cpu.txt)
[report-arm-2cpu.txt](https://github.com/gchq/sleeper/files/9960034/report-arm-2cpu.txt)
[report-x86-1cpu.txt](https://github.com/gchq/sleeper/files/9960035/report-x86-1cpu.txt)
[report-x86-2cpu.txt](https://github.com/gchq/sleeper/files/9960036/report-x86-2cpu.txt)